meson.build

```diff
@@ -166,7 +167,7 @@ python = pymod.find_installation('python3', modules: doxygen.found() and get_opt
 rst2html = find_program('rst2html', required: get_option('html-docs'))
 rst2pdf = find_program('rst2pdf', required: get_option('pdf-docs'))
 git = find_program('git', required: false)
-if git.found()
+if git.found() and fs.is_dir('.git')
 	ret = run_command(git, 'rev-parse', '--short', '--revs-only', 'HEAD', check: true)
 	cdata.set_quoted('REVISION', ret.stdout().strip())
 else
```

Comment:
So what about something like this (untested at all, and I barely know Meson):
```meson
revision = '-1'
if git.found():
    ret = run_command(git, 'rev-parse', '--short', '--revs-only', 'HEAD', check: false)
    if ret.return_code() == 0:
        revision = ret.stdout().strip()
cdata.set_quoted('REVISION', revision)
```

Reply:
Looks OK.

Reply:
I recommend to use `fs.exists()` since the `.git` filesystem entry can be a file, in which case the contents of the file are a single line:
```
gitdir: path/to/.git/worktrees/foobar
```

This is how `git worktree` works.